automaton.py

```diff
@@ -1,37 +1,140 @@
+import enum
+
 class Transition():
 
-    def __init__(self, dest: str, target: str, symbol: str):
+    def __init__(self, dest: str, target: list=[], symbol: list=[]):
```

Comment:
This is incorrect, `source`, `target` and `symbol` should be of type `str` and mandatory for a transition instance